pkg/scaffold/resource/options.go

```diff
@@ -0,0 +1,234 @@
+	}
+	if len(opts.Kind) == 0 {
+		return fmt.Errorf(kindRequired)
+	}
+
+	// Check if the Group has a valid DNS1123 subdomain value
+	if err := IsDNS1123Subdomain(opts.Group); err != nil {
+		return fmt.Errorf("group name is invalid: (%v)", err)
+	}
+
+	// Check if the version follows the valid pattern
+	if !versionRegex.MatchString(opts.Version) {
+		return fmt.Errorf(
+			"version must match %s (was %s)", versionPattern, opts.Version)
+	}
+
+	// Check if the Kind is PascalCase
+	if opts.Kind != flect.Pascalize(opts.Kind) {
+		return fmt.Errorf("kind must be PascalCase (expected %s was %s)", flect.Pascalize(opts.Kind), opts.Kind)
+	}
+
+	// TODO: validate plural and resource strings if provided
+
+	return nil
+}
+
+// safeImport returns a cleaned version of the provided string that can be used for imports
+func (opts *Options) safeImport(unsafe string) string {
+	safe := unsafe
+
+	// Remove dashes and dots
+	safe = strings.Replace(safe, "-", "", -1)
+	safe = strings.Replace(safe, ".", "", -1)
+
+	return safe
+}
+
+func (opts *Options) NewV1Resource(repo, domain string) *Resource {
+	res := opts.newResource()
+
+	// Use the k8s.io/api package for core resources
+	coreGroups := map[string]string{
+		"apps":                  "",
+		"admissionregistration": "k8s.io",
+		"apiextensions":         "k8s.io",
+		"authentication":        "k8s.io",
+		"autoscaling":           "",
+		"batch":                 "",
+		"certificates":          "k8s.io",
+		"core":                  "",
+		"extensions":            "",
+		"metrics":               "k8s.io",
+		"policy":                "",
+		"rbac.authorization":    "k8s.io",
+		"storage":               "k8s.io",
+	}
+
+	pkg := path.Join(repo, "pkg", "apis", res.GroupPackageName, opts.Version)
+	// TODO: remove the dependency on the created file to differentiate local and external resources
+	// as this requires the Resource to be recreated after scaffolding the file
+	file := filepath.Join("pkg", "apis", res.GroupPackageName, opts.Version, fmt.Sprintf("%s_types.go", strings.ToLower(opts.Kind)))
+	if _, err := os.Stat(file); os.IsNotExist(err) {
+		if coreDomain, found := coreGroups[opts.Group]; found {
+			pkg = path.Join("k8s.io", "api", opts.Group, opts.Version)
+			domain = coreDomain
+		}
+		// TODO: need to support '--resource-pkg-path' flag for specifying resourcePath
+	}
+	res.Package = pkg
+	res.Domain = opts.Group
+	if domain != "" {
+		res.Domain += "." + domain
+	}
+
+	res.CreateExampleReconcileBody = opts.CreateExampleReconcileBody
+
+	return res
+}
+
+func (opts *Options) NewResource(repo, domain string) *Resource {
+	res := opts.newResource()
+
+	// Use the k8s.io/api package for core resources
+	coreGroups := map[string]string{
+		"admission":             "k8s.io",
+		"admissionregistration": "k8s.io",
+		"apps":                  "",
+		"auditregistration":     "k8s.io",
+		"apiextensions":         "k8s.io", // NOTE: where is this?
+		"authentication":        "k8s.io",
```

Comment:
Remove the comments? 
Why should we add them in the code?

Reply:
Oh, I didn't mean to get that committed. I just wanted to ask about those. I guess this is as good place as any other to ask about it. @mengqiy @DirectXMan12 

I was looking if these map required and updated. I checked in https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/api and I found a different set of apis. Is that the right place to look at?

The found differences are:
- `apiextensions` and `metric` are not subpackages of `k8s.io/api`
- `rbac.authorization` is not a subpackage either, but `rbac` is
- `setting` is not a subpackage either, but `settings` is
- There were two additional subpackages, `discovery` and `flowcontrol`
Should these reflect into changes to the map somehow?